### PR TITLE
account for aliased attributes when attribute methods are mixed in

### PR DESCRIPTION
**Pull request by joelvh**
Fixes https://github.com/state-machines/state_machines-activemodel/issues/27

Moves `alias_attribute` fixes from https://github.com/state-machines/state_machines-activerecord/pull/95

### CODE REVIEW

**Comment by joelvh** [comment]
@seuros here's the ActiveModel implementation

**Comment by joelvh** [comment]
@seuros tests pass locally - I can't see details of the build error.

**Comment by joelvh** [comment]
@seuros are you interested in merging this rather than https://github.com/state-machines/state_machines-activerecord/pull/95?

**Comment by joelvh** [comment]
@seuros just checking to see if this is the route to go or the ActiveRecord PR.

**Comment by joelvh** [comment]
@seuros will this be merged?

**Comment by seuros** [comment]
Yes, sorry. I'm back at OSSing.

File: lib/state_machines/integrations/active_model.rb [hunk 1]
```diff
@@ -407,6 +407,10 @@ def around_validation(object)
       def define_state_initializer
         define_helper :instance, <<-end_eval, __FILE__, __LINE__ + 1
           def initialize(params = {})
+            params.transform_keys! do |key|
+              self.class.attribute_aliases[key.to_s] || key.to_s
+            end if self.class.respond_to?(:attribute_aliases)
```
Comment (seuros) [inline lib/state_machines/integrations/active_model.rb:412]: is the condition for old AM version ?   We will support only 6.0+

**Review by seuros — APPROVED** [review]

**Comment by joelvh** [comment]
hi @seuros, thanks for merging this!

Will [this](https://github.com/state-machines/state_machines-activerecord/pull/95) also be merged?

**Pull request merged**